projects/media-viewer/src/lib/media-viewer.component.ts

```diff
@@ -50,13 +52,33 @@ enum ConvertibleContentTypes {
 })
 export class MediaViewerComponent implements OnChanges, OnDestroy, AfterContentInit {
 
+  @ViewChild('viewerRef') viewerRef: ElementRef;
+
   @Input() url;
   @Input() downloadFileName: string;
   @Input() contentType: string;
 
   @Input() showToolbar = true;
   @Input() toolbarButtonOverrides: any = {};
-  @Input() height = 'calc(100vh - 42px)';
+
+  @Input()
+  public set height(value: string) {
+    this._givenHeight = value;
+  }
+
+  public get height(): string {
```

Comment:
I think it would be better if we don't use the getter function in this way, because we're using the field in the template and it's going to be called every time the angular change detection runs. when in reality, the value only changes on zoom, rotate or when the input height changes
so using a function that only gets called at those times would be better in terms of performance

Reply:
That's all true. I just wanted to avoid introducing a new property, for example internalHeight or something like this, which can be consumed by the template and updated when height input param changes.

Reply:
but that would probably be better in terms of performance
there are quite a few instances where we have done this in the app, and we're trying to get rid of them

Reply:
Removed setter/getter for height input param and implement AfterViewChecked hook to track component location change. Tried to male it lean and not to trigger change detection on every run.